### PR TITLE
File share lease client updates internal leaseid

### PR DESCRIPTION
**Pull request by jaschrep-msft**

### CODE REVIEW

**Review by kasobol-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Release History
 
 ## 12.9.0-beta.4 (Unreleased)
+- ShareLeaseClient now updates it's leaseID through a lease change.
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-file-share/CHANGELOG.md:4]: Should we say lease modification? since it applies to change and renew?

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-share/CHANGELOG.md:4]: and acquire

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/CHANGELOG.md (hunk 1 above)
Comment (jaschrep-msft) [reply sdk/storage/azure-storage-file-share/CHANGELOG.md:4]: I mean, that technically was the behavior already. Acquire and renew didn't technically need the changes but if we ever changed how these clients were initialized they would.

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-share/CHANGELOG.md:4]: ok, I think that makes sense

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**